rdmo/projects/managers.py

```diff
@@ -52,6 +52,46 @@ def filter_catalogs(self, catalogs=None, exclude_catalogs=None, exclude_null=Tru
             catalogs_filter &= ~Q(catalog__in=exclude_catalogs)
         return self.filter(catalogs_filter)
 
+    def filter_groups(self, groups):
+        if not groups:
+            return self
+
+        # need to import here to prevent circular import
+        # queryset methods need to be refactored in modules otherwise
+        from django.contrib.auth import get_user_model
+
+        from rdmo.projects.models import Membership
+
+        # users in the given groups
+        users = get_user_model().objects.filter(groups__in=groups)
+        # memberships for those users with role 'owner'
+        memberships = Membership.objects.filter(role='owner', user__in=users)
+        # projects that have those memberships
+        return self.filter(memberships__in=memberships).distinct()
+
+    def filter_projects_for_task_or_view(self, instance):
+
+        # if View/Task is not available it should not show for any project
+        if not instance.available:
+            return self.none()
+
+        # projects that have an unavailable catalog should be disregarded
+        qs = self.filter(catalog__available=True)
```

Comment:
this is maybe still needs a design decision.
Currently, when a project has a catalog that is unavailable it will be disregarded for syncing. 
Is this a typical use-case or does the sync needs to go over all applicable projects anyway? @m6121